typescript/interchain/contracts/InterchainAccountRouter.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import {OwnableMulticall, Call} from "./OwnableMulticall.sol";
+
+// ============ External Imports ============
+import {Router} from "@abacus-network/app/contracts/Router.sol";
+import {TypeCasts} from "@abacus-network/core/contracts/libs/TypeCasts.sol";
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+/*
+ * @title The Hello World App
+ * @dev You can use this simple app as a starting point for your own application.
+ */
+contract InterchainAccountRouter is Router {
+    bytes constant bytecode = type(OwnableMulticall).creationCode;
+    bytes32 constant bytecodeHash = bytes32(keccak256(bytecode));
+
+    constructor(
+        address _abacusConnectionManager,
+        address _interchainGasPaymaster
+    ) {
+        // Transfer ownership of the contract to deployer
+        _transferOwnership(msg.sender);
+        // Set the addresses for the ACM and IGP
+        // Alternatively, this could be done later in an initialize method
+        _setAbacusConnectionManager(_abacusConnectionManager);
+        _setInterchainGasPaymaster(_interchainGasPaymaster);
+    }
+
+    function dispatch(uint32 _destinationDomain, Call[] calldata calls)
+        external
+    {
+        _dispatch(_destinationDomain, abi.encode(msg.sender, calls));
+    }
+
+    function getInterchainAccount(uint32 _origin, address _sender)
+        public
+        view
+        returns (address)
+    {
+        return _getInterchainAccount(_salt(_origin, _sender));
+    }
+
+    function getDeployedInterchainAccount(uint32 _origin, address _sender)
+        public
```

Comment:
Why not internal?

Reply:
skipping to avoid changing bytecode from deployed